### ISSUE TITLE
Why isn't it / How is it possible to vectorize sound incrementally?

### ISSUE THREAD

**Issue by dims12** [pytorch/fairseq#2635]
I was wishing to vectorize sound incrementally as it sounds in real time, can't find how.

I was thinking if I take some small portion of sound and pass it to feature extractor, I will get the same 512 feature vector which I was getting first from complete sound. I was thinking it is either some number of milliseconds from article but it didn't work. Then I wrote code to find the window size but failed.

#### Code

```
import torch
from fairseq.models.wav2vec import Wav2VecModel
import numpy as np

cp = torch.load('wav2vec_large.pt')
model = Wav2VecModel.build_model(cp['args'], task=None)
model.load_state_dict(cp['model'])
model.eval()

max_window_size = 1000
wav_complete = torch.randn(1, max_window_size)
vectors_complete = model.feature_extractor(wav_complete)
vectors_complete = vectors_complete.detach().numpy().reshape(512, -1)

for window_size in range(2, max_window_size + 1, 1):
    wav_part = wav_complete[0, 0:window_size].reshape(1, -1)

    wav_shape = tuple(wav_part.shape)

    try:
        vectors_partial = model.feature_extractor(wav_part)
        vectors_partial = vectors_partial.detach().numpy().reshape(512, -1)

        partial_shape = tuple(vectors_partial.shape)
        vectors_complete_part = vectors_complete[:, 0:partial_shape[1]]

        comp = (vectors_complete_part == vectors_partial).all(axis=0)

        if np.any(comp):
            print("%s\t%s\t%d of %d " % (wav_shape, partial_shape, np.sum(comp), comp.size))

    except:
        print("%s\tERROR" % str(wav_shape))

    pass

```

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): master
 - PyTorch Version (e.g., 1.0): 1.6.0
 - OS (e.g., Linux): Linux
 - How you installed fairseq (`pip`, source): source
 - Build command you used (if compiling from source): according to manual 
 - Python version: 3.8.5
 - CUDA/cuDNN version: 10.2
 - GPU models and configuration: GTX 1070
 - Any other relevant information:

**Comment by dims12** [comment]
The cause is probably group normalization and it is probably hardcoded (ex. vec 1)

https://github.com/pytorch/fairseq/blob/master/fairseq/models/wav2vec/wav2vec.py#L436
